src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/config/S3Configuration.kt

```diff
@@ -1,34 +1,71 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.config
 
-import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
 import software.amazon.awssdk.regions.Region
 import software.amazon.awssdk.services.s3.S3Client
 import java.net.URI
 
+data class S3Bucket(val enabled: Boolean, val bucketName: String, val client: S3Client)
+
+data class S3BucketConfiguration(
+  val enabled: Boolean,
+  val provider: String,
+  val region: String,
+  val accessKeyId: String,
+  val secretAccessKey: String,
+  val endpoint: Endpoint,
+  val bucket: Bucket,
+) {
+  data class Endpoint(
+    val uri: String
+  )
+
+  data class Bucket(
+    val name: String
+  )
+}
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "aws.s3")
+class S3Buckets(
+  val storage: S3BucketConfiguration,
+  val ndmis: S3BucketConfiguration,
```

Comment:
Personally, I usually prefer things to be introduced when they are first used; if for any reason we drop the reporting-via-jobs work, this change is now in and `ndmisS3Bucket()` is unused.

So I'd prefer the NDMIS work to _add_ the NDMIS-related wiring in, rather than being here (plus tests, etc.)

But this is not a blocker

Reply:
agree - the only issue is that it's not obvious how this change really works unless there are two sets of configuration. that's the main reason i added it.